### PR TITLE
Ramazan Ince NodeJs HW2

### PR DESCRIPTION
**Pull request by RamazanInce**

### CODE REVIEW

File: week2/homework/src/index.js [hunk 1]
```diff
@@ -1,3 +1,68 @@
 'use strict';
 
-// TODO: Write the homework code in this file
+let fs = require('fs');
+let args = process.argv.slice(2);
+// console.log(args);
```
Comment (eidosam) [inline week2/homework/src/index.js]: Commented code can be removed

File: week2/homework/src/index.js [hunk 2]
```diff
@@ -1,3 +1,68 @@
...
+
+let command = args[0];
+let todoItem = args[1];
+
```
Comment (eidosam) [inline week2/homework/src/index.js]: `fs`, `args`, `command` and `todoItem` are not re-assigned, so they have to be declared with `const`

File: week2/homework/src/index.js [hunk 3]
```diff
@@ -1,3 +1,68 @@
...
+if (command === undefined || 'help') {
```
Comment (eidosam) [inline week2/homework/src/index.js]: This is always evaluated to `true`. You can read it this way

    if ((command === undefined) || 'help') {
      // ...
    }

It should be re-written

    if (command === undefined || command === 'help') {
      // ...
    }

Or ` (!command || command === 'help') `

File: week2/homework/src/index.js [hunk 4]
```diff
@@ -1,3 +1,68 @@
...
+      } else console.error(error);
+    } else {
+      console.log(todoList);
+    }
+  });
+}
+
+function add(todoItem) {
+  fs.appendFile('./todoList.txt', todoItem + '\n', error => {
+    console.error(error);
```
Comment (eidosam) [inline week2/homework/src/index.js]: The `error` should be checked, and logged only if it is present. So the reporting to user will be proper.
Like

    if(error) return console.error(error);
    return console.log('Todo item is added');

File: week2/homework/src/index.js [hunk 5]
```diff
@@ -1,3 +1,68 @@
...
+  });
+}
+
+function remove(index) {
+  fs.readFile('./todoList.txt', 'utf8', (err, data) => {
+    if (err) throw err;
+    let newList = data.split('\n');
```
Comment (eidosam) [inline week2/homework/src/index.js]: `newList` is not re-assigned, so it can be declared with `const`

File: week2/homework/src/index.js [hunk 6]
```diff
@@ -1,3 +1,68 @@
...
+    newList.splice(index - 1, 1);
+    fs.writeFile('./todoList.txt', newList.join('\n'), err => {
+      if (err) throw err;
+    });
+  });
+}
+
+function reset(todoItem) {
+  fs.writeFile('./todoList.txt', '', function(err) {
```
Comment (eidosam) [inline week2/homework/src/index.js]: Use arrow-functions where possible.

File: week2/homework/src/index.js [hunk 7]
```diff
@@ -1,3 +1,68 @@
...
+    if (err) throw err;
+    console.log('Reset!');
+  });
+}
+
+function help(todoItem) {
+  fs.readFile('./help.txt', 'utf8', function(err, data) {
```
Comment (eidosam) [inline week2/homework/src/index.js]: Use arrow-functions where possible.

File: week2/homework/src/index.js (hunk 3 above)
Comment (RamazanInce) [reply week2/homework/src/index.js]: I did not try that one, I did it in the last step, Thanks for the correction.

File: week2/homework/src/index.js (hunk 4 above)
Comment (RamazanInce) [reply week2/homework/src/index.js]: that is ok, thanks

**Comment by wouterkleijn** [comment]
Thanks @eidosam for addressing the comments. Thanks @RamazanInce for implementing the necessary changes :)

**Pull request closed without merging**